haas/api.py

```diff
@@ -529,24 +626,24 @@ def network_create(network, creator, access, net_id):
     auth_backend = get_auth_backend()
     _assert_absent(model.Network, network)
 
-    # Check authorization and legality of arguments, and find correct 'access' and 'creator'
-    if creator != "admin":
-        creator = _must_find(model.Project, creator)
-        auth_backend.require_project_access(creator)
+    # Check authorization and legality of arguments, and find correct 'access' and 'owner'
+    if owner != "admin":
+        owner = _must_find(model.Project, owner)
+        auth_backend.require_project_access(owner)
         # Project-owned network
-        if access != creator.label:
-            raise BadArgumentError("Project-created networks must be accessed only by that project.")
+        if access != owner.label:
+            raise BadArgumentError("Project-owned networks must be accessed only by that project.")
```

Comment:
It's probably fine to leave it as for now, but the error message needs to change somehow. Perhaps "Project-owned networks must be accessible by the owner."

Fyi, to recieve non-string arguments, you'd need to explicitly specify a schema (these will be mandatory soon anyway). Read the comments in `haas.rest` and the docs for the schema library:

https://pypi.python.org/pypi/schema